core/src/cluster_nodes.rs

```diff
@@ -291,6 +307,32 @@ fn get_nodes(cluster_info: &ClusterInfo, stakes: &HashMap<Pubkey, u64>) -> Vec<N
     .collect()
 }
 
+// root     : [0]
+// 1st layer: [1, 2, ..., fanout]
+// 2nd layer: [[fanout + 1, ..., fanout * 2],
+//             [fanout * 2 + 1, ..., fanout * 3],
+//             ...
+//             [fanout * fanout + 1, ..., fanout * (fanout + 1)]]
+// 3rd layer: ...
+// ...
```

Comment:
Could we add a comment describing:
1) How the shreds are actually propagated in these layers
2) What the significance of the return value is (nodes in the same neighborhood within the layer the input `index` is in)

Reply:
Might also be good to add a comment about how this is different from the existing `get_retransmit_peers()` function, maybe `This function returns only the peers and no children, but a todo in the future. Once the feature is enabled, this function will replace the other one`

Reply:
Added more comments describing the propagation path.
The returned values are literally "re-transmit peers" which are nodes in the _next layer_.